src/component/panels/RangesPanel/RangesHeader.tsx

```diff
@@ -137,16 +138,11 @@ function RangesHeader({
   function saveAsHTMLHandler() {
     if (Array.isArray(ranges?.values) && ranges.values.length > 0) {
       const { originFrequency: observedFrequency, nucleus } = info;
-
-      const nbDecimalDelta = getNumberOfDecimals(
-        rangesPreferences.deltaPPM.format,
-      );
-      const nbDecimalJ = getNumberOfDecimals(rangesPreferences.deltaHz.format);
-
+      const currentPreferences = preferences.nuclei[nucleus.toLowerCase()];
       const result = rangesToACS(ranges.values, {
         nucleus, // '19f'
-        nbDecimalDelta, // 2
-        nbDecimalJ, // 1
+        nbDecimalDelta: currentPreferences.ppm,
+        nbDecimalJ: currentPreferences.hz,
```

Comment:
I have some notes:

1- we format them based on the delta in ppm and Hz from the panel preferences which is per nucleus.

![Screenshot 2023-09-19 at 09 26 14](https://github.com/cheminfo/nmrium/assets/35760236/7caae2fe-5095-4c1c-9886-23ed63dcf007)

why do you need to take the format from the general preferences?

Reply:
As i know the`nbDecimalDelta` and `nbDecimalJ` are accepting a number of decimals and not the format

Reply:
it was a confusion based on the description of the ticket #2676, also the use of deltaHz for coupling decimals is wrong because we have Coupling(Hz)

Reply:
does it accept both `number` or `string`?